docs/index.md

```diff
@@ -88,34 +68,102 @@ Currently, Rust is the best-supported language for writing SpacetimeDB modules.
 
 SpacetimeDB was designed first and foremost as the backend for multiplayer Unity games. To learn more about using SpacetimeDB with Unity, jump on over to the [SpacetimeDB Unity Tutorial](/docs/unity/part-1).
 
+## Key architectural concepts
+
+### Host
+A SpacetimeDB **host** is a combination of a database and server that runs [modules](#module). You can run your own SpacetimeDB host, or use a public host.
+Public hosts typically charge [energy](#energy) to run modules with a large amount of data or clients. However, the SpacetimeDB testnet is free.
+
+### Module
+A SpacetimeDB **module** is an application that runs on a [host](#host).
+
+A module exports [tables](#table), which store data, and [reducers](#reducer), which allow [clients](#client) to make requests.
+
+Technically, a SpacetimeDB module is a [WebAssembly module](https://developer.mozilla.org/en-US/docs/WebAssembly) that imports a specific low-level [WebAssembly API](/webassembly-api) and exports a small number of special functions. However, the SpacetimeDB [server-side libraries](#server-side-libraries) hide these low-level details. As a developer, writing a module is mostly like writing any other C# or Rust application, except for the fact that a [special CLI tool](/install) is used to build and deploy the application.
+
+### Table
+A SpacetimeDB **table** is a database table. Tables are declared in a module's native language. For instance, in Rust, a table is declared like so:
+
+```rust
+#[spacetimedb::table(name = person, public)]
+pub struct Player {
+   id: u64,
+   name: String,
+   age: u32,
+   user: Identity,
+}
+```
+
+The contents of a table can be read by [reducers](#reducer).
+Tables marked `public` can also be read by [clients](#client).
+
+### Reducer
+A **reducer** is a function exported by a [module](#module).
+Connected [clients](#client-side-sdks) can call reducers to interact with the module.
+This is a form of [remote procedure call](https://en.wikipedia.org/wiki/Remote_procedure_call).
+Reducers can be invoked across languages. For example, a Rust [module](#module) can export a reducer like so:
+
+```rust
+#[spacetimedb::reducer]
+pub fn set_player_name(ctx: &spacetimedb::ReducerContext, id: u64, name: String) -> Result<(), String> {
+   // ...
+}
+```
+
+And a C# [client](#client) can call that reducer:
+
+```cs
+void Main() {
+   // ...setup code, then...
+   Reducer.SetPlayerName(57, "Marceline");
+}
+```
+
+These look mostly like regular function calls, but under the hood, they are cross-language RPC calls.
```

Comment:
```suggestion
These look mostly like regular function calls, but under the hood, the client sends a request which the module processes and responds to.
```

I think we probably don't want to use acronyms like "RPC" too often. It's good to give readers who know what RPCs are a hook to understand, but it's not good to make readers who don't know go learn in order to understand.